src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue28986.cs

```diff
@@ -13,6 +13,85 @@ public Issue28986(TestDevice device) : base(device)
+		// there should be NO bottom padding from the navigation bar.
+		App.WaitForElement("ContentGrid");
+
+		// First, set to None to get the baseline position (no safe area padding)
+		App.Tap("GridResetNoneButton");
+		var noneRect = App.WaitForElement("MainGrid").GetRect();
+
+		// Now set bottom edge to SoftInput (keyboard is still hidden)
+		App.Tap("GridSetBottomSoftInputButton");
+
+		// Wait for layout to update
+		App.WaitForElement("MainGrid");
+
+		// Get the current settings to verify SoftInput is set
+		var currentSettings = App.FindElement("CurrentSettings").GetText();
+		Assert.That(currentSettings, Does.Contain("Bottom:SoftInput"),
+			"Bottom edge should be set to SoftInput");
+
+		// Get the rect after setting SoftInput
+		var softInputRect = App.WaitForElement("MainGrid").GetRect();
+
+		// Verify that the bottom position is the same as None (no padding applied)
+		// The height should be the same because SoftInput should not add padding when keyboard is hidden
+		Assert.That(softInputRect.Height, Is.EqualTo(noneRect.Height).Within(5),
+			"MainGrid height should be the same with SoftInput as with None when keyboard is hidden (no bottom padding)");
+
+		// Also verify the Y position hasn't shifted
+		Assert.That(softInputRect.Y, Is.EqualTo(noneRect.Y).Within(5),
+			"MainGrid Y position should be the same with SoftInput as with None when keyboard is hidden");
+	}
+
+	[Test]
+	[Category(UITestCategories.SafeAreaEdges)]
+	public void AllRegionStillAppliesBottomPaddingWhenKeyboardHidden()
+	{
+		// This test validates that the fix for #31870 doesn't break SafeAreaRegions.All
+		// When SafeAreaEdges is set to All, it should respect safe area behavior consistently.
+		// The specific assertion depends on whether the device has a navigation bar.
+		App.WaitForElement("ContentGrid");
+
+		// First, set to None to get the baseline position (no safe area padding)
+		App.Tap("GridResetNoneButton");
+		var noneRect = App.WaitForElement("MainGrid").GetRect();
+
+		// Set bottom edge to SoftInput to test the specific behavior we fixed
+		App.Tap("GridSetBottomSoftInputButton");
+		App.WaitForElement("MainGrid");
+		var softInputRect = App.WaitForElement("MainGrid").GetRect();
+
+		// Now set to All (should apply all safe area insets)
+		App.Tap("GridResetAllButton");
+		App.WaitForElement("MainGrid");
+
+		// Get the current settings to verify All is set
+		var currentSettings = App.FindElement("CurrentSettings").GetText();
+		Assert.That(currentSettings, Does.Contain("All (Full safe area)"),
+			"SafeAreaEdges should be set to All");
+
+		// Get the rect after setting All
+		var allRect = App.WaitForElement("MainGrid").GetRect();
+
+		// The key validation: All should have same or less height than None (never more)
+		// And All should behave differently than SoftInput when keyboard is hidden
+		Assert.That(allRect.Height, Is.LessThanOrEqualTo(noneRect.Height),
```

Comment:
Add a comment clarifying why `LessThanOrEqualTo` is used (to avoid future maintainers reverting it).